Domain/ConfigurationExtensions.cs

```diff
@@ -8,6 +8,25 @@
 
 namespace Microsoft.Its.Domain
 {
+    /// <summary>
+    /// Deserializes a string into an object.
+    /// </summary>
+    /// <param name="input">The string to deserialize.</param>
+    /// <param name="type">Optional type to deserialize as.</param>
+    /// <returns>
+    /// The deserialized object.
+    /// </returns>
+    public delegate object CustomDeserialize(string input, Type type = null);
```

Comment:
I'd suggest `SerializeEvent` and `DeserializeEvent` as names.

Reply:
I'll make the change.